docs/development/sidekiq_flows.md

```diff
@@ -0,0 +1,143 @@
+# Sidekiq flows - submission process notes
+
+## Processing of requests to HmrcInterface
+
+### Workers and services flow summary
+```text
+- BulkSubmissionsWorker (plural)
+  - BulkSubmissionWorker (singular)
+   - BulkSubmissionService
+    - HmrcInterfaceBulkSubmissionWorker (singular)
+      - HmrcInterfaceSubmissionWorker (on uc-one-submission queue)
+        - HmrcInterfaceSubmissionService
+         - HmrcInterfaceResultsWorker (retry upto x times)
+          - HmrcInterfaceResultService
+      - HmrcInterfaceSubmissionWorker (on uc-two-submission queue)
+        - HmrcInterfaceSubmissionService
+         - HmrcInterfaceResultsWorker (retry upto x times)
+          - HmrcInterfaceResultService
+```
+
+### Workers and services flow
+
+* BulkSubmissionsWorker (plural) called by scheduled job
+    This finds all pending bulk submissions and calls the BulkSubmissionWorker (singular) for each
+
+* BulkSubmissionsWorker (plural) calls BulkSubmissionWorker (singular) for all pending bulk submissions
+
+* BulkSubmissionWorker (singular) calls BulkSubmissionService.call for each bulk_submission
+
+* BulkSubmissionService performs the following:
+  - updates bulk submission status as :preparing
+  - for each record in bulk submission's attached original_file
+    - create use case one submission record with client details, start and end date, status: pending
+    - create use case two submission record with client details, start and end date, status: pending
+ - updates bulk submission status as :prepared
+ - enqueues HmrcInterfaceBulkSubmissionWorker for the bulk_submission
+
+* HmrcInterfaceBulkSubmissionWorker job , is performed at 9pm, calling HmrcInterfaceBulkSubmissionWorker (note the singular) for each bulk submission in a pending state.
+
+* HmrcInterfaceBulkSubmissionWorker (singular) identifies pending submissions associated with that bulk submission and calls HmrcInterfaceSubmissionWorker for each submission
+
+* HmrcInterfaceSubmissionWorker
+
+ - enqueues HmrcInterfaceSubmissionService on uc-one-submission
```

Comment:
I think some of the bulleting/indenting is wrong here

Reply:
done, when testing locally at least